blocks/extensions/saveload.py

```diff
@@ -94,12 +74,20 @@ def do(self, callback_name, *args):
             path = self.path
             if from_user:
                 path, = from_user
-            secure_dump(self.main_loop, path, use_cpickle=self.use_cpickle)
-            filenames = self.save_separately_filenames(path)
-            for attribute in self.save_separately:
-                secure_dump(getattr(self.main_loop, attribute),
-                            filenames[attribute], cPickle.dump,
-                            protocol=DEFAULT_PROTOCOL)
+            if self.save_separately:
+                to_add = {}
+                for attr in self.save_separately:
+                    to_add[attr] = getattr(self.main_loop, attr)
+            else:
+                to_add = None
```

Comment:
You can also put `to_add = None` before `if self.save_separately`, thereby further reducing the number of lines.